frontend/src/views/omni/Modals/UpdateKernelArgs.vue

```diff
@@ -0,0 +1,198 @@
+<!--
+Copyright (c) 2025 Sidero Labs, Inc.
+
+Use of this software is governed by the Business Source License
+included in the LICENSE file.
+-->
+<script setup lang="ts">
+import { computed, onMounted, type Ref, ref } from 'vue'
+import { useRoute, useRouter } from 'vue-router'
+
+import { Runtime } from '@/api/common/omni.pb'
+import { Code } from '@/api/google/rpc/code.pb'
+import { type Resource, ResourceService } from '@/api/grpc'
+import { type KernelArgsSpec, type KernelArgsStatusSpec } from '@/api/omni/specs/omni.pb'
+import { withRuntime } from '@/api/options'
+import { DefaultNamespace, KernelArgsStatusType, KernelArgsType } from '@/api/resources'
+import Watch from '@/api/watch'
+import TButton from '@/components/common/Button/TButton.vue'
+import TInput from '@/components/common/TInput/TInput.vue'
+import TAlert from '@/components/TAlert.vue'
+import { showError, showSuccess } from '@/notification'
+import CloseButton from '@/views/omni/Modals/CloseButton.vue'
+
+const args = ref('')
+const route = useRoute()
+const router = useRouter()
+
+const status: Ref<Resource<KernelArgsStatusSpec> | undefined> = ref()
+const argsWatch = new Watch(status)
+
+argsWatch.setup({
+  runtime: Runtime.Omni,
+  resource: {
+    id: route.query.machine as string,
+    namespace: DefaultNamespace,
+    type: KernelArgsStatusType,
+  },
+})
+
+const currentArgs = computed(() => {
+  return status.value?.spec.current_args?.join(' ') || ''
+})
+const currentCmdline = computed(() => {
+  return status.value?.spec.current_cmdline || ''
+})
+const unmetConditions = computed(() => {
+  return status.value?.spec.unmet_conditions || []
+})
+
+const md = {
+  id: route.query.machine as string,
+  namespace: DefaultNamespace,
+  type: KernelArgsType,
+}
+
+onMounted(async () => {
+  try {
+    const current = await ResourceService.Get<Resource<KernelArgsSpec>>(
+      md,
+      withRuntime(Runtime.Omni),
+    )
+
+    args.value = (current.spec.args || []).map((arg) => arg.trim()).join(' ')
+  } catch (e) {
+    if (e.code !== Code.NOT_FOUND) {
+      throw e
+    }
+  }
+})
+
+const handleUpdateKernelArgs = async () => {
+  try {
+    await updateKernelArgs()
+  } catch (e) {
+    showError(`Failed to Update Kernel Args`, e.message)
+
+    return
+  }
+
+  showSuccess(`Kernel args are updated`)
+
+  close()
+}
+
+const updateKernelArgs = async () => {
+  let existing: Resource<KernelArgsSpec> | undefined = undefined
+  try {
+    existing = await ResourceService.Get<Resource<KernelArgsSpec>>(md, withRuntime(Runtime.Omni))
```

Comment:
dont we have it already? from the `onMounted` call? if it might have changed since then, should we `watch` instead of `get`? or is there something else at play here?